src/signalrclient/callback_manager.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+#include "callback_manager.h"
+
+namespace signalr
+{
+    callback_manager::callback_manager(const web::json::value& dtor_clear_arguments)
+        : m_dtor_clear_arguments(dtor_clear_arguments)
+    { }
+
+    callback_manager::~callback_manager()
+    {
+        clear(m_dtor_clear_arguments);
+    }
+
+    int callback_manager::register_callback(const std::function<void(const web::json::value&)>& callback)
+    {
+        auto callback_id = m_id++;
+
+        {
+            std::lock_guard<std::mutex> lock(m_map_lock);
+
+            m_callbacks.insert(std::make_pair(callback_id, callback));
+        }
+
+        return callback_id;
+    }
+
+    bool callback_manager::complete_callback(int callback_id, const web::json::value& arguments)

```

Comment:
This might also benefit from comments. Complte_Callback removes the callback from the manager and invokes it one last time, correct? That's not obvious from the name.